File: serde-generate/runtime/csharp/Serde/Unit.cs

using System;

namespace Serde
{
    public sealed class Unit : IEquatable<Unit>
    {
        public Unit() { }

        public override bool Equals(object obj) => obj is Unit unit;

        public bool Equals(Unit other) => other != null;

        public override int GetHashCode() => 793253941;
    }
}

File: serde-generate/runtime/swift/Serde/Unit.swift
//  Copyright © Diem Association. All rights reserved.

import Foundation

public struct Unit: Equatable {
  public func Equals(obj:Any) -> Bool {
    return (obj as? Unit) != nil
  }
  
  public static func ==(l:Unit, r:Unit) -> Bool {
    return true
  }
  
  public static func !=(l:Unit, r:Unit) -> Bool {
    return false
  }
  
  public func GetHashCode() -> Int { return 793253941 }
}
